cmd/limactl/list.go

```diff
@@ -220,15 +240,35 @@ func listAction(cmd *cobra.Command, args []string) error {
 			options.TerminalWidth = w
 		}
 	}
-	// --yq implies --format json unless --format yaml has been explicitly specified
+
+	// --yq implies --format json unless --format has been explicitly specified
 	if len(yq) != 0 && !cmd.Flags().Changed("format") {
 		format = "json"
 	}
+
 	// Always pipe JSON and YAML through yq to colorize it if isTTY
 	if len(yq) == 0 && (format == "json" || format == "yaml") {
 		yq = append(yq, ".")
 	}
 
+	for _, f := range filter {
+		// only allow fields, ==, !=, and literals.
+		valid := regexp.MustCompile(`^[a-zA-Z0-9_.\s"'-=><!]+$`)
+		if !valid.MatchString(f) {
+			return fmt.Errorf("unsafe characters in filter expression: %q", f)
+		}
```

Comment:
I don't think this is useful. The whole point of `--filter` and `--yq` is to provide YQ expressions that are going to be evaluated. There is no vulnerability; the expression is provided by the user themselves.

And we already get an error when the YQ expression is invalid, e.g.

```console
❯ l ls -l '.vmType =='
FATA[0000] failed to apply filter select(.vmType ==): '==' expects 2 args but there is 1
```

So I don't understand the purpose of this "validation", especially since it might make legitimate use cases invalid (e.g. you are restricting which characters are allowed inside literal strings).